### PR TITLE
Update CRUD Tests 

### PR DESCRIPTION
**Pull request by jlord**
This updates the CRUD tests in these ways:

- Fixes `bulkWrite` to pass on `arrayFilters`
- Fixes `insertMany` to return `insertedIds` in the correct format
- Adds tests for `bulkWrite`, `insertOne` and `insertMany`
- Updates the tests files from the [specification directory](https://github.com/mongodb/specifications/tree/master/source/crud/tests).

For review ease, skip the first commit which dumps the new files in ✨ 

🍍 

NODE-1228

### CODE REVIEW

**Comment by mbroadst** [comment]
@jlord and I spoke offline about this, my general comments are that we need update the inline documentation, and also update the migration guide indicating that this is a breaking change

File: lib/collection.js [hunk 1]
```diff
@@ -724,7 +714,7 @@ define.classMethod('bulkWrite', { callback: true, promise: true });
  * @typedef {Object} Collection~insertWriteOpResult
  * @property {Number} insertedCount The total amount of documents inserted.
  * @property {object[]} ops All the documents inserted using insertOne/insertMany/replaceOne. Documents contain the _id field if forceServerObjectId == false for insertOne/insertMany
- * @property {ObjectId[]} insertedIds All the generated _id's for the inserted documents.
+ * @property {ObjectId[]} insertedIds Map of the index of the inserted document to the id of the inserted document.
```
Comment (daprahamian) [inline lib/collection.js]: We should change the jsdoc type to `{Object.<string, ObjectId}`

Comment (mbroadst) [reply lib/collection.js]: agreed, but I think its actually `{Object.<Number, ObjectId>}`

Comment (daprahamian) [reply lib/collection.js]: Right, that :)

Comment (jlord) [reply lib/collection.js]: 👍

**Comment by mbroadst** [comment]
@jlord this needs to be rebased

**Pull request merged**